snap/info.go

```diff
@@ -139,6 +173,7 @@ type SideInfo struct {
 	EditedDescription string   `yaml:"description,omitempty" json:"description,omitempty"`
 	Private           bool     `yaml:"private,omitempty" json:"private,omitempty"`
 	Paid              bool     `yaml:"paid,omitempty" json:"paid,omitempty"`
+	InstanceKey       string   `yaml:"instance-key,omitempty" json:"instance-key,omitempty"`
```

Comment:
mmh, in a way it's a bit of a convenient but strange place to store this,  it could also go in Info and be added there when we read installed snap data from disk. The reason to do it this way is mostly it avoids needing to change SnapSetup as well.